heat/core/tests/test_manipulations.py

```diff
@@ -202,3 +201,53 @@ def test_squeeze(self):
             data.squeeze(axis='y')
         with self.assertRaises(ValueError):
             ht.argmin(data, axis=-4)
+
+    def test_unique(self):
+        size = ht.MPI_WORLD.size
+        rank = ht.MPI_WORLD.rank
+        torch_array = torch.arange(size, dtype=torch.int32).expand(size, size)
+        split_zero = ht.array(torch_array, split=0)
+
+        exp_axis_none = ht.arange(size, dtype=ht.int32)
+        res = ht.unique(split_zero, sorted=True)
+        self.assertTrue((res._DNDarray__array == exp_axis_none._DNDarray__array).all())
+
+        exp_axis_zero = ht.arange(size, dtype=ht.int32).expand_dims(0)
+        res = ht.unique(split_zero, sorted=True, axis=0)
+        self.assertTrue((res._DNDarray__array == exp_axis_zero._DNDarray__array).all())
+
+        exp_axis_one = ht.array([rank], dtype=ht.int32).expand_dims(0)
+        split_zero_transposed = ht.array(torch_array.transpose(0, 1), split=0)
+        res = ht.unique(split_zero_transposed, sorted=True, axis=1)
+        self.assertTrue((res._DNDarray__array == exp_axis_one._DNDarray__array).all())
+
+        split_one = ht.array(torch_array, dtype=ht.int32, split=1)
+
+        exp_axis_none = ht.arange(size, dtype=ht.int32)
+        res = ht.unique(split_one, sorted=True)
+        self.assertTrue((res._DNDarray__array == exp_axis_none._DNDarray__array).all())
+
+        exp_axis_zero = ht.array([rank], dtype=ht.int32).expand_dims(0)
+        res = ht.unique(split_one, sorted=True, axis=0)
+        self.assertTrue((res._DNDarray__array == exp_axis_zero._DNDarray__array).all())
+
+        exp_axis_one = ht.array([rank] * size, dtype=ht.int32).expand_dims(1)
+        res = ht.unique(split_one, sorted=True, axis=1)
+        self.assertTrue((res._DNDarray__array == exp_axis_one._DNDarray__array).all())
+
+        # torch_array = torch.tensor([
```

Comment:
These tests will be uncommented as soon as Allgatherv() works? Is this reason that the code coverage is also not hitting all the lines?

We should postpone the merge for right after the Allgatherv() fix or issue another ticket.

Reply:
I changed the code to only use Allgatherv() with axis=0, so everything works as expected